### ISSUE TITLE
Incorrect properties lifting to the parent node

### ISSUE THREAD

**Issue by Pluralia** [langium/langium#296]
Rules `AssignmentExpression`, `OrExpression` and `AndExpression` have a common property `operator` and inherit `BinaryExpression`, so the `BinaryExpression` in the generated AST interface has to contain a property `operator: 'Or' | 'And' | ':'`. However, type of the `operator` is `'Or' | 'And'`, because it omits an operator of the `AssignmentExpression` rule.

```
Model:
    elements+=OrAndAssignmentExpression*;

OrAndAssignmentExpression returns Expression:
    OrExpression ({AssignmentExpression.left=current} operator=':' right=OrExpression)*;

// Dummy rule to make AssignmentExpression inherit from BinaryExpression
AssignmentExpression returns BinaryExpression:
    {AssignmentExpression} left=ID operator=':' right=OrExpression;

OrExpression returns Expression:
	AndExpression ({BinaryExpression.left=current} operator='Or' right=AndExpression)*;

AndExpression returns Expression:
	ValueExpression ({BinaryExpression.left=current} operator='And' right=ValueExpression)*;

ValueExpression returns Expression:
	{BoolValue} value=BOOL | {IDValue} value=ID;

terminal ID:				   /\$?[_a-zA-Z][\w_]*/;
terminal BOOL returns boolean: /(true)|(false)/;
```

Langium version: 0.2.1
Package name: langium-cli

## Steps To Reproduce

1. Generate an AST with the grammar above
2. Check out the `BinaryExpression` rule in the AST

## The current behavior

Type of `operator` from `BinaryExpression` doesn't contain ':'
```
export interface BinaryExpression extends Expression {
    operator: 'Or' | 'And'
    ...
}
```

## The expected behavior

Type of `operator` from `BinaryExpression` contain ':'
```
export interface BinaryExpression extends Expression {
    operator: 'Or' | 'And' | ':'
    ...
}
```

**Comment by pluralia** [comment]
Can be resolved by types declaration. Depends on #510.

**Comment by pluralia** [comment]
@msujew The problem in the example appears in the dummy rule -- we could try to replace it with declared types and solve the problem. Declared types with validation implemented in #510 are more suitable for this goal.

We also should investigate the need to solve the problem for the case where similar constructions are not used as a dummy rule, but I think it would be good to open a new issue with better example for this case.
